internal/publisher/kafka.go

```diff
@@ -41,10 +62,24 @@ func (p *KafkaPublisher) Close() error {
 	return p.producer.Close()
 }
 
+func (p *KafkaPublisher) keyData(e *Event) []byte {
+	if data, ok := e.Data[p.cfg.MessageKeyFrom]; ok {
+		key, _ := json.Marshal(data)
+		return key
+	}
+
+	return []byte(e.Table)
+}
+
 // NewProducer return new Kafka producer instance.
 func NewProducer(pCfg *config.PublisherCfg) (sarama.SyncProducer, error) {
 	cfg := sarama.NewConfig()
 	cfg.Producer.Partitioner = sarama.NewRandomPartitioner
+
+	if pCfg.IsMessageKeyExists() {
+		cfg.Producer.Partitioner = sarama.NewHashPartitioner
```

Comment:
Would it be possible for the user to somehow select the partitioner using the configuration? This one is great for java guys, but personnally I actually need the `NewConsistentCRCHashPartitioner` instead since I have (probably too much) librdkafka (used by most of Confluent clients) producers in production